### PR TITLE
Add `useFractions` parameter to the timemap rendering

### PR DESCRIPTION
**Pull request by lpugin**
Can be triggered with:
```
--timemap-options "{'includeMeasures': true, 'includeRests': true , 'useFractions': true}"
```

Example:
image
(tuplet-001.mei in https://www.verovio.org/test-suite.xhtml?cat=tuplet)

<details>

```json
[
	{
		"measureOn": "m84asto",
		"on": [
			"n1",
			"n127qmvy" 
		],
		"qstamp": [
			0,
			1 
		],
		"tempo": "120.000000",
		"tstamp": 0 
	},
	{
		"off": [
			"n1" 
		],
		"on": [
			"nba5ezt" 
		],
		"qstamp": [
			3,
			14 
		],
		"tstamp": 107 
	},
	{
		"off": [
			"nba5ezt" 
		],
		"on": [
			"n1x8ts9s" 
		],
		"qstamp": [
			3,
			7 
		],
		"tstamp": 214 
	},
	{
		"off": [
			"n127qmvy" 
		],
		"on": [
			"ndszjqa" 
		],
		"qstamp": [
			1,
			2 
		],
		"tstamp": 250 
	},
	{
		"off": [
			"n1x8ts9s" 
		],
		"on": [
			"njl3bz5" 
		],
		"qstamp": [
			9,
			14 
		],
		"tstamp": 321 
	},
	{
		"off": [
			"njl3bz5" 
		],
		"on": [
			"n6co8e5" 
		],
		"qstamp": [
			6,
			7 
		],
		"tstamp": 429 
	},
	{
		"off": [
			"ndszjqa" 
		],
		"on": [
			"nko2pgp" 
		],
		"qstamp": [
			1,
			1 
		],
		"tstamp": 500 
	},
	{
		"off": [
			"n6co8e5" 
		],
		"on": [
			"n15jh1sh" 
		],
		"qstamp": [
			15,
			14 
		],
		"tstamp": 536 
	},
	{
		"off": [
			"n15jh1sh" 
		],
		"on": [
			"n1xby4ad" 
		],
		"qstamp": [
			9,
			7 
		],
		"tstamp": 643 
	},
	{
		"off": [
			"n1xby4ad",
			"nko2pgp" 
		],
		"on": [
			"n1bncvfm",
			"nb83egn" 
		],
		"qstamp": [
			3,
			2 
		],
		"tstamp": 750 
	},
	{
		"off": [
			"n1bncvfm" 
		],
		"on": [
			"ndjbgk1" 
		],
		"qstamp": [
			45,
			28 
		],
		"tstamp": 804 
	},
	{
		"off": [
			"ndjbgk1" 
		],
		"on": [
			"ncr6gns" 
		],
		"qstamp": [
			12,
			7 
		],
		"tstamp": 857 
	},
	{
		"off": [
			"ncr6gns" 
		],
		"on": [
			"nurs0nr" 
		],
		"qstamp": [
			51,
			28 
		],
		"tstamp": 911 
	},
	{
		"off": [
			"nurs0nr" 
		],
		"on": [
			"n6ix7z6" 
		],
		"qstamp": [
			27,
			14 
		],
		"tstamp": 964 
	},
	{
		"off": [
			"nb83egn" 
		],
		"on": [
			"n18xjgtu" 
		],
		"qstamp": [
			2,
			1 
		],
		"tstamp": 1000 
	},
	{
		"off": [
			"n6ix7z6" 
		],
		"on": [
			"nicdme5" 
		],
		"qstamp": [
			57,
			28 
		],
		"tstamp": 1018 
	},
	{
		"off": [
			"nicdme5" 
		],
		"on": [
			"n13b4swe" 
		],
		"qstamp": [
			15,
			7 
		],
		"tstamp": 1071 
	},
	{
		"off": [
			"n13b4swe" 
		],
		"on": [
			"nqwil7n" 
		],
		"qstamp": [
			9,
			4 
		],
		"tstamp": 1125 
	},
	{
		"off": [
			"nqwil7n" 
		],
		"on": [
			"nqy9ov5" 
		],
		"qstamp": [
			33,
			14 
		],
		"tstamp": 1179 
	},
	{
		"off": [
			"nqy9ov5" 
		],
		"on": [
			"nrdaa9w" 
		],
		"qstamp": [
			69,
			28 
		],
		"tstamp": 1232 
	},
	{
		"off": [
			"n18xjgtu" 
		],
		"on": [
			"ndim9hy" 
		],
		"qstamp": [
			5,
			2 
		],
		"tstamp": 1250 
	},
	{
		"off": [
			"nrdaa9w" 
		],
		"on": [
			"n1ypknaj" 
		],
		"qstamp": [
			18,
			7 
		],
		"tstamp": 1286 
	},
	{
		"off": [
			"n1ypknaj" 
		],
		"on": [
			"nby10lj" 
		],
		"qstamp": [
			75,
			28 
		],
		"tstamp": 1339 
	},
	{
		"off": [
			"nby10lj" 
		],
		"on": [
			"n19ayqyu" 
		],
		"qstamp": [
			39,
			14 
		],
		"tstamp": 1393 
	},
	{
		"off": [
			"n19ayqyu" 
		],
		"on": [
			"n21" 
		],
		"qstamp": [
			81,
			28 
		],
		"tstamp": 1446 
	},
	{
		"measureOn": "mctjjnr",
		"off": [
			"n21",
			"ndim9hy" 
		],
		"on": [
			"n28",
			"nhmzvbh" 
		],
		"qstamp": [
			3,
			1 
		],
		"tstamp": 1500 
	},
	{
		"off": [
			"nhmzvbh" 
		],
		"on": [
			"n1dpr7b1" 
		],
		"qstamp": [
			33,
			10 
		],
		"tstamp": 1650 
	},
	{
		"off": [
			"n28" 
		],
		"on": [
			"n29" 
		],
		"qstamp": [
			7,
			2 
		],
		"tstamp": 1750 
	},
	{
		"off": [
			"n1dpr7b1" 
		],
		"on": [
			"n10ryly4" 
		],
		"qstamp": [
			18,
			5 
		],
		"tstamp": 1800 
	},
	{
		"off": [
			"n10ryly4" 
		],
		"on": [
			"n1dddq94" 
		],
		"qstamp": [
			39,
			10 
		],
		"tstamp": 1950 
	},
	{
		"off": [
			"n1dddq94" 
		],
		"on": [
			"n10z0zq9" 
		],
		"qstamp": [
			21,
			5 
		],
		"tstamp": 2100 
	},
	{
		"off": [
			"n10z0zq9" 
		],
		"on": [
			"n1tods6l" 
		],
		"qstamp": [
			9,
			2 
		],
		"tstamp": 2250 
	},
	{
		"off": [
			"n29" 
		],
		"on": [
			"n1w1ps48" 
		],
		"qstamp": [
			5,
			1 
		],
		"tstamp": 2500 
	},
	{
		"off": [
			"n1w1ps48" 
		],
		"on": [
			"n15zqrvd" 
		],
		"qstamp": [
			11,
			2 
		],
		"tstamp": 2750 
	},
	{
		"off": [
			"n15zqrvd",
			"n1tods6l" 
		],
		"qstamp": [
			6,
			1 
		],
		"tstamp": 3000 
	} 
] 
```
</details>

Without `useFractions`:

<details>

```json
[
	{
		"measureOn": "mwujfgu",
		"on": [
			"n1",
			"nngz0ja" 
		],
		"qstamp": 0,
		"tempo": "120.000000",
		"tstamp": 0 
	},
	{
		"off": [
			"n1" 
		],
		"on": [
			"nwfj1p3" 
		],
		"qstamp": 0.2142857142857143,
		"tstamp": 107 
	},
	{
		"off": [
			"nwfj1p3" 
		],
		"on": [
			"ntxtlr0" 
		],
		"qstamp": 0.4285714285714285,
		"tstamp": 214 
	},
	{
		"off": [
			"nngz0ja" 
		],
		"on": [
			"nogqdm2" 
		],
		"qstamp": 0.5,
		"tstamp": 250 
	},
	{
		"off": [
			"ntxtlr0" 
		],
		"on": [
			"n1f5sv1g" 
		],
		"qstamp": 0.6428571428571429,
		"tstamp": 321 
	},
	{
		"off": [
			"n1f5sv1g" 
		],
		"on": [
			"n1ozb87d" 
		],
		"qstamp": 0.8571428571428571,
		"tstamp": 429 
	},
	{
		"off": [
			"nogqdm2" 
		],
		"on": [
			"n1yujvb4" 
		],
		"qstamp": 1,
		"tstamp": 500 
	},
	{
		"off": [
			"n1ozb87d" 
		],
		"on": [
			"n4owge9" 
		],
		"qstamp": 1.071428571428571,
		"tstamp": 536 
	},
	{
		"off": [
			"n4owge9" 
		],
		"on": [
			"n1vmrto9" 
		],
		"qstamp": 1.285714285714286,
		"tstamp": 643 
	},
	{
		"off": [
			"n1vmrto9",
			"n1yujvb4" 
		],
		"on": [
			"n145f64p",
			"n26xz3o" 
		],
		"qstamp": 1.5,
		"tstamp": 750 
	},
	{
		"off": [
			"n145f64p" 
		],
		"on": [
			"n9wqsok" 
		],
		"qstamp": 1.607142857142857,
		"tstamp": 804 
	},
	{
		"off": [
			"n9wqsok" 
		],
		"on": [
			"n150o3dm" 
		],
		"qstamp": 1.714285714285714,
		"tstamp": 857 
	},
	{
		"off": [
			"n150o3dm" 
		],
		"on": [
			"n14q50p5" 
		],
		"qstamp": 1.821428571428571,
		"tstamp": 911 
	},
	{
		"off": [
			"n14q50p5" 
		],
		"on": [
			"nxy0ljr" 
		],
		"qstamp": 1.928571428571429,
		"tstamp": 964 
	},
	{
		"off": [
			"n26xz3o" 
		],
		"on": [
			"n1yiiniv" 
		],
		"qstamp": 2,
		"tstamp": 1000 
	},
	{
		"off": [
			"nxy0ljr" 
		],
		"on": [
			"nvli7y1" 
		],
		"qstamp": 2.035714285714286,
		"tstamp": 1018 
	},
	{
		"off": [
			"nvli7y1" 
		],
		"on": [
			"nxs1b10" 
		],
		"qstamp": 2.142857142857143,
		"tstamp": 1071 
	},
	{
		"off": [
			"nxs1b10" 
		],
		"on": [
			"n11u7k72" 
		],
		"qstamp": 2.25,
		"tstamp": 1125 
	},
	{
		"off": [
			"n11u7k72" 
		],
		"on": [
			"nmqvdbh" 
		],
		"qstamp": 2.357142857142857,
		"tstamp": 1179 
	},
	{
		"off": [
			"nmqvdbh" 
		],
		"on": [
			"n1fh64e1" 
		],
		"qstamp": 2.464285714285714,
		"tstamp": 1232 
	},
	{
		"off": [
			"n1yiiniv" 
		],
		"on": [
			"nlo1siy" 
		],
		"qstamp": 2.5,
		"tstamp": 1250 
	},
	{
		"off": [
			"n1fh64e1" 
		],
		"on": [
			"n19vow6r" 
		],
		"qstamp": 2.571428571428572,
		"tstamp": 1286 
	},
	{
		"off": [
			"n19vow6r" 
		],
		"on": [
			"n1xorcaz" 
		],
		"qstamp": 2.678571428571428,
		"tstamp": 1339 
	},
	{
		"off": [
			"n1xorcaz" 
		],
		"on": [
			"n1dwrkm7" 
		],
		"qstamp": 2.785714285714286,
		"tstamp": 1393 
	},
	{
		"off": [
			"n1dwrkm7" 
		],
		"on": [
			"n21" 
		],
		"qstamp": 2.892857142857143,
		"tstamp": 1446 
	},
	{
		"measureOn": "m1tqejxq",
		"off": [
			"n21",
			"nlo1siy" 
		],
		"on": [
			"n28",
			"n16ti28b" 
		],
		"qstamp": 3,
		"tstamp": 1500 
	},
	{
		"off": [
			"n16ti28b" 
		],
		"on": [
			"n1lpmyi0" 
		],
		"qstamp": 3.3,
		"tstamp": 1650 
	},
	{
		"off": [
			"n28" 
		],
		"on": [
			"n29" 
		],
		"qstamp": 3.5,
		"tstamp": 1750 
	},
	{
		"off": [
			"n1lpmyi0" 
		],
		"on": [
			"nyfbw2w" 
		],
		"qstamp": 3.6,
		"tstamp": 1800 
	},
	{
		"off": [
			"nyfbw2w" 
		],
		"on": [
			"nyzm7hl" 
		],
		"qstamp": 3.9,
		"tstamp": 1950 
	},
	{
		"off": [
			"nyzm7hl" 
		],
		"on": [
			"n1d6m50k" 
		],
		"qstamp": 4.2,
		"tstamp": 2100 
	},
	{
		"off": [
			"n1d6m50k" 
		],
		"on": [
			"n1m94u5f" 
		],
		"qstamp": 4.5,
		"tstamp": 2250 
	},
	{
		"off": [
			"n29" 
		],
		"on": [
			"n1nid92e" 
		],
		"qstamp": 5,
		"tstamp": 2500 
	},
	{
		"off": [
			"n1nid92e" 
		],
		"on": [
			"n6edrzs" 
		],
		"qstamp": 5.5,
		"tstamp": 2750 
	},
	{
		"off": [
			"n6edrzs",
			"n1m94u5f" 
		],
		"qstamp": 6,
		"tstamp": 3000 
	} 
] 
```
</details>

### CODE REVIEW

**Comment by craigsapp** [comment]
A separate issue in the timemap is that tempo values are strings instead of numbers (probably was already like that):

```json
	"tempo": "120.000000",
```

More preferable:

```json
	"tempo": 120.000000,
```

Using `strstream`/`ostream` instead of `printf`/`tostring` can remove trailing zeros:

```cpp
double tempo = 120.0;
stringstream t;
t << tempo;
cout << "\"tempo\":" << t.str() << "," << endl;
t.str(""); // clear contents of stringstream when in a loop
```

```cpp
ostream output;
double tempo = 120.0;
output << "\"tempo\":" << tempo << "," << endl;
```

which should produce a more friendly:

```json
	"tempo": 120,
```

I am encoding the example in Humdrum for comparison of Fraction outputs between verovio and humlib.

**Comment by lpugin** [comment]
The new timemap with:
* The `qstamp` renamed to `qfrac` with `useFractions`
* The tempo as number

<details>

```json
[
	{
		"measureOn": "m18gezl5",
		"on": [
			"n1",
			"no3wxed" 
		],
		"qfrac": [
			0,
			1 
		],
		"tempo": 120,
		"tstamp": 0 
	},
	{
		"off": [
			"n1" 
		],
		"on": [
			"nb0neuf" 
		],
		"qfrac": [
			3,
			14 
		],
		"tstamp": 107 
	},
	{
		"off": [
			"nb0neuf" 
		],
		"on": [
			"nhxuh3z" 
		],
		"qfrac": [
			3,
			7 
		],
		"tstamp": 213 
	},
	{
		"off": [
			"no3wxed" 
		],
		"on": [
			"n1caig9p" 
		],
		"qfrac": [
			1,
			2 
		],
		"tstamp": 249 
	},
	{
		"off": [
			"nhxuh3z" 
		],
		"on": [
			"nrqdpb7" 
		],
		"qfrac": [
			9,
			14 
		],
		"tstamp": 320 
	},
	{
		"off": [
			"nrqdpb7" 
		],
		"on": [
			"n1tmrrj7" 
		],
		"qfrac": [
			6,
			7 
		],
		"tstamp": 427 
	},
	{
		"off": [
			"n1caig9p" 
		],
		"on": [
			"n1lvfnyv" 
		],
		"qfrac": [
			1,
			1 
		],
		"tstamp": 498 
	},
	{
		"off": [
			"n1tmrrj7" 
		],
		"on": [
			"nbkk43x" 
		],
		"qfrac": [
			15,
			14 
		],
		"tstamp": 533 
	},
	{
		"off": [
			"nbkk43x" 
		],
		"on": [
			"nuhk3f4" 
		],
		"qfrac": [
			9,
			7 
		],
		"tstamp": 640 
	},
	{
		"off": [
			"nuhk3f4",
			"n1lvfnyv" 
		],
		"on": [
			"nt0znjx",
			"n76ay34" 
		],
		"qfrac": [
			3,
			2 
		],
		"tstamp": 747 
	},
	{
		"off": [
			"nt0znjx" 
		],
		"on": [
			"n1cbpcvh" 
		],
		"qfrac": [
			45,
			28 
		],
		"tstamp": 800 
	},
	{
		"off": [
			"n1cbpcvh" 
		],
		"on": [
			"nvoedv2" 
		],
		"qfrac": [
			12,
			7 
		],
		"tstamp": 854 
	},
	{
		"off": [
			"nvoedv2" 
		],
		"on": [
			"n1vrcewf" 
		],
		"qfrac": [
			51,
			28 
		],
		"tstamp": 907 
	},
	{
		"off": [
			"n1vrcewf" 
		],
		"on": [
			"nboiysp" 
		],
		"qfrac": [
			27,
			14 
		],
		"tstamp": 960 
	},
	{
		"off": [
			"n76ay34" 
		],
		"on": [
			"n1ac76pd" 
		],
		"qfrac": [
			2,
			1 
		],
		"tstamp": 996 
	},
	{
		"off": [
			"nboiysp" 
		],
		"on": [
			"n2x5kuc" 
		],
		"qfrac": [
			57,
			28 
		],
		"tstamp": 1014 
	},
	{
		"off": [
			"n2x5kuc" 
		],
		"on": [
			"nklcmt1" 
		],
		"qfrac": [
			15,
			7 
		],
		"tstamp": 1067 
	},
	{
		"off": [
			"nklcmt1" 
		],
		"on": [
			"n1hzalpy" 
		],
		"qfrac": [
			9,
			4 
		],
		"tstamp": 1120 
	},
	{
		"off": [
			"n1hzalpy" 
		],
		"on": [
			"n1sffdrl" 
		],
		"qfrac": [
			33,
			14 
		],
		"tstamp": 1174 
	},
	{
		"off": [
			"n1sffdrl" 
		],
		"on": [
			"n17h2ivc" 
		],
		"qfrac": [
			69,
			28 
		],
		"tstamp": 1227 
	},
	{
		"off": [
			"n1ac76pd" 
		],
		"on": [
			"n1b7ipep" 
		],
		"qfrac": [
			5,
			2 
		],
		"tstamp": 1245 
	},
	{
		"off": [
			"n17h2ivc" 
		],
		"on": [
			"nr5vqj9" 
		],
		"qfrac": [
			18,
			7 
		],
		"tstamp": 1280 
	},
	{
		"off": [
			"nr5vqj9" 
		],
		"on": [
			"nblm861" 
		],
		"qfrac": [
			75,
			28 
		],
		"tstamp": 1334 
	},
	{
		"off": [
			"nblm861" 
		],
		"on": [
			"n1mn0mxl" 
		],
		"qfrac": [
			39,
			14 
		],
		"tstamp": 1387 
	},
	{
		"off": [
			"n1mn0mxl" 
		],
		"on": [
			"n21" 
		],
		"qfrac": [
			81,
			28 
		],
		"tstamp": 1440 
	},
	{
		"measureOn": "m1waok9i",
		"off": [
			"n21",
			"n1b7ipep" 
		],
		"on": [
			"n28",
			"n112ziz6" 
		],
		"qfrac": [
			3,
			1 
		],
		"tstamp": 1494 
	},
	{
		"off": [
			"n112ziz6" 
		],
		"on": [
			"nff7fs3" 
		],
		"qfrac": [
			33,
			10 
		],
		"tstamp": 1643 
	},
	{
		"off": [
			"n28" 
		],
		"on": [
			"n29" 
		],
		"qfrac": [
			7,
			2 
		],
		"tstamp": 1743 
	},
	{
		"off": [
			"nff7fs3" 
		],
		"on": [
			"nzf9r52" 
		],
		"qfrac": [
			18,
			5 
		],
		"tstamp": 1793 
	},
	{
		"off": [
			"nzf9r52" 
		],
		"on": [
			"n1ub9jvl" 
		],
		"qfrac": [
			39,
			10 
		],
		"tstamp": 1942 
	},
	{
		"off": [
			"n1ub9jvl" 
		],
		"on": [
			"nwys7jj" 
		],
		"qfrac": [
			21,
			5 
		],
		"tstamp": 2091 
	},
	{
		"off": [
			"nwys7jj" 
		],
		"on": [
			"n1seywy7" 
		],
		"qfrac": [
			9,
			2 
		],
		"tstamp": 2241 
	},
	{
		"off": [
			"n29" 
		],
		"on": [
			"n4r6hzo" 
		],
		"qfrac": [
			5,
			1 
		],
		"tstamp": 2490 
	},
	{
		"off": [
			"n4r6hzo" 
		],
		"on": [
			"n241vg9" 
		],
		"qfrac": [
			11,
			2 
		],
		"tstamp": 2739 
	},
	{
		"off": [
			"n241vg9",
			"n1seywy7" 
		],
		"qfrac": [
			6,
			1 
		],
		"tstamp": 2988 
	} 
] 
```
</details>

**Comment by craigsapp** [comment]
The test timemap matches my calculations in humlib (and Humdrum Extras) tools:

Screenshot 2024-10-08 at 12 14 46 AM

<details>
<summary> Click to view test Humdrum data and timing extractions </summary>

```tsv
**tstamp	**qstamp	**qfrac	**kern	**kern
*	*	*	*clefF4	*clefG2
*	*	*	*k[f#c#g#d#a#]	*k[f#c#g#d#a#]
*	*	*	*M3/4	*M3/4
*MM120	*MM120	*MM120	*MM120	*MM120
=1	=1	=1	=1	=1
0	0	0	8AA#L	(56%3fff#LL
107	0.214286	3/14	.	56%3eee#
214	0.428571	3/7	.	56%3ddd#
250	0.5	1/2	8F#	.
321	0.642857	9/14	.	56%3ccc#
429	0.857143	6/7	.	56%3bb
500	1	1	8f#	.
536	1.07143	15/14	.	56%3ddd#
643	1.28571	9/7	.	56%3aa#JJ
750	1.5	3/2	8G#X	112%3gg##LLL
804	1.60714	45/28	.	112%3ff#
857	1.71429	12/7	.	112%3ee#
911	1.82143	51/28	.	112%3dd#
964	1.92857	27/14	.	112%3cc##
1000	2	2	8B	.
1018	2.03571	57/28	.	112%3ee#
1071	2.14286	15/7	.	112%3dd#
1125	2.25	9/4	.	112%3cc#
1179	2.35714	33/14	.	112%3b#
1232	2.46429	69/28	.	112%3a#
1250	2.5	5/2	8F#J	.
1286	2.57143	18/7	.	112%3g##
1339	2.67857	75/28	.	112%3f#
1393	2.78571	39/14	.	112%3e#
1446	2.89286	81/28	.	112%3d#JJJ)
=2	=2	=2	=2	=2
1500	3	3	40%3AA#L	(8c##
1650	3.3	33/10	40%3E#	.
1750	3.5	7/2	.	4.a#)
1800	3.6	18/5	40%3A#	.
1950	3.9	39/10	40%3c##	.
2100	4.2	21/5	40%3E#J	.
2250	4.5	9/2	4.e#	.
2500	5	5	.	8bL
2750	5.5	11/2	.	8b#J
=	=	=	=	=
*-	*-	*-	*-	*-
```

</details>

**Review by craigsapp — APPROVED** [review]

File: include/vrv/timemap.h [hunk 1]
```diff
@@ -15,6 +15,12 @@
 
 //----------------------------------------------------------------------------
 
+#include "horizontalaligner.h"
```
Comment (ahankinson) [inline include/vrv/timemap.h:18]: I still find it a bit weird that `Fraction` is defined in `horizontalaligner`… I think I know why (fractions are mostly used for time, and time is usually on the horizontal) but it’s still a bit strange to be talking about “horizontal alignment” in a MIDI context.

**Review by ahankinson — APPROVED** [review]

**Comment by craigsapp** [comment]
I think that the horizontal alignment is used for aligning `<layer>` elements by `@dur` information, and this will later be converted to MIDI ticks and create the timemap, @lpugin?

**Comment by lpugin** [comment]
What we could do is rename `timemap.h/cpp` to `time.h/cpp` and move `Fraction` there. Would that be less weird? Or would you necessarily expect a `fraction.h/cpp`?

**Comment by craigsapp** [comment]
I would be more expecting `fraction.h/cpp`, but you mentioned that you want to reserve files for MEI element classes.  If it is a support class for a single other class, it could be placed inside the class that uses it (only useful if there is a single class that uses it).   Or maybe in vrv.h/cpp?

**Comment by lpugin** [comment]
OK. See #3822

**Pull request merged**